vulkan/vulkan.hpp

```diff
@@ -16914,5 +16914,89 @@ namespace VULKAN_HPP_NAMESPACE
+  constexpr uint32_t MaxMemoryHeaps                   = VK_MAX_MEMORY_HEAPS;
+  constexpr uint32_t MaxMemoryTypes                   = VK_MAX_MEMORY_TYPES;
+  constexpr uint32_t MaxPhysicalDeviceNameSize        = VK_MAX_PHYSICAL_DEVICE_NAME_SIZE;
+  constexpr uint32_t MaxShaderModuleIdentifierSizeExt = VK_MAX_SHADER_MODULE_IDENTIFIER_SIZE_EXT;
+  constexpr uint32_t QueueFamilyExternal              = VK_QUEUE_FAMILY_EXTERNAL;
+  constexpr uint32_t QueueFamilyForeignExt            = VK_QUEUE_FAMILY_FOREIGN_EXT;
+  constexpr uint32_t QueueFamilyIgnored               = VK_QUEUE_FAMILY_IGNORED;
+  constexpr uint32_t Remaining3DSlicesExt             = VK_REMAINING_3D_SLICES_EXT;
+  constexpr uint32_t RemainingArrayLayers             = VK_REMAINING_ARRAY_LAYERS;
+  constexpr uint32_t RemainingMipLevels               = VK_REMAINING_MIP_LEVELS;
+  constexpr uint32_t ShaderUnusedKhr                  = VK_SHADER_UNUSED_KHR;
+  constexpr uint32_t SubpassExternal                  = VK_SUBPASS_EXTERNAL;
+  constexpr uint32_t True                             = VK_TRUE;
+  constexpr uint32_t UuidSize                         = VK_UUID_SIZE;
+  constexpr uint64_t WholeSize                        = VK_WHOLE_SIZE;
+  constexpr uint32_t HeaderVersion                    = VK_HEADER_VERSION;
+
+  consteval auto ApiVersionMajor( auto const version )
+  {
+    return ( ( (uint32_t)( version ) >> 22U ) & 0x7FU );
+  }
+
+  consteval auto ApiVersionMinor( auto const version )
+  {
+    return ( ( (uint32_t)( version ) >> 12U ) & 0x3FFU );
+  }
+
+  consteval auto ApiVersionPatch( auto const version )
+  {
+    return ( (uint32_t)(version)&0xFFFU );
+  }
+
+  consteval auto ApiVersionVariant( auto const version )
+  {
+    return ( (uint32_t)( version ) >> 29U );
+  }
+
+  consteval auto MakeApiVersion( auto const variant, auto const major, auto const minor, auto const patch )
+  {
+    return ( ( ( (uint32_t)( variant ) ) << 29U ) | ( ( (uint32_t)( major ) ) << 22U ) | ( ( (uint32_t)( minor ) ) << 12U ) | ( (uint32_t)( patch ) ) );
+  }
+
+  [[deprecated( "This define is deprecated. VK_MAKE_API_VERSION should be used instead." )]] consteval auto
+    MakeVersion( auto const major, auto const minor, auto const patch )
+  {
+    return ( ( ( (uint32_t)( major ) ) << 22U ) | ( ( (uint32_t)( minor ) ) << 12U ) | ( (uint32_t)( patch ) ) );
+  }
+
+  [[deprecated( "This define is deprecated. VK_API_VERSION_MAJOR should be used instead." )]] consteval auto VersionMajor( auto const version )
+  {
+    return ( (uint32_t)( version ) >> 22U );
+  }
+
+  [[deprecated( "This define is deprecated. VK_API_VERSION_MINOR should be used instead." )]] consteval auto VersionMinor( auto const version )
+  {
+    return ( ( (uint32_t)( version ) >> 12U ) & 0x3FFU );
+  }
+
+  [[deprecated( "This define is deprecated. VK_API_VERSION_PATCH should be used instead." )]] consteval auto VersionPatch( auto const version )
```

Comment:
deprecated is only available since C++14

Reply:
`VULKAN_HPP_DEPRECATED`

Reply:
I presume I should now also define `VULKAN_HPP_CONSTEVAL`. I'll get on that.

Reply:
Okay. I have turned them into templated `constexpr` functions using SFINAE. I would have liked to use concepts, but unlike `vulkan.ixx` we cannot assume C++20 is available.